Packs/AbnormalSecurity/Integrations/AbnormalSecurity/AbnormalSecurity.py

```diff
@@ -172,6 +197,65 @@ def submit_false_positive_report_request(self, portal_link):
 
         return response
 
+    def get_a_list_of_vendors_request(self, page_size, page_number):
+        params = assign_params(pageSize=page_size, pageNumber=page_number)
+
+        headers = self._headers
+
+        response = self._http_request("get", "vendors", params=params, headers=headers)
+
+        response = self._remove_keys_from_response(response, ["pageNumber"])
+
+        return response
+
+    def get_the_details_of_a_specific_vendor_request(self, vendorDomain):
+        headers = self._headers
+
+        response = self._http_request("get", f"vendors/{vendorDomain}/details", headers=headers)
+
+        return response
+
+    def get_the_activity_of_a_specific_vendor_request(self, vendorDomain):
+        headers = self._headers
+
+        response = self._http_request("get", f"vendors/{vendorDomain}/activity", headers=headers)
+
+        return response
+
+    def get_a_list_of_vendor_cases_request(self, filter_, page_size, page_number):
+        params = assign_params(filter=filter_, pageSize=page_size, pageNumber=page_number)
+
+        headers = self._headers
+
+        response = self._http_request("get", "vendor-cases", params=params, headers=headers)
+
+        response = self._remove_keys_from_response(response, ["pageNumber"])
+
+        return response
+
+    def get_the_details_of_a_vendor_case_request(self, caseId):
+        headers = self._headers
+
+        response = self._http_request("get", f"vendor-cases/{caseId}", headers=headers)
+
+        return response
+
+    def get_a_list_of_unanalyzed_abuse_mailbox_campaigns_request(self, start, end):
+        params = assign_params(start=start, end=end)
+
+        headers = self._headers
+
+        response = self._http_request("get", "abuse_mailbox/not_analyzed", params=params, headers=headers)
+
+        return response
+
+    def _remove_keys_from_response(self, response, keys_to_remove):
```

Comment:
For now I think this should suffice - did not want to add more complexity w/out necessity (so no need for a recursive pop here)